### ISSUE TITLE
codegen:generate - queries to local storage

### ISSUE THREAD

**Issue by Vlad-Shevliakov** [apollographql/apollo-tooling#2237]
Hey!
I have a small question how to generate types for queries that are not in `schema.graphql` because they are local to `InMemoryCache`.

Here is my script that generates types:
```json
{
    "apollo:schema": "apollo client:download-schema --endpoint=http://localhost:8080/ schema.graphql",
    "apollo:gen": "apollo codegen:generate --localSchemaFile=schema.graphql --target=typescript --includes='src/**' --tagName=gql _types_"
}
```

And I have a request that will be directed to the local storage and it is **naturally** not in the scheme.
```js
export const ADD_PRODUCT_TO_CART = gql`
  mutation AddProductToCart($id: String!) {
    addToCart(id: $id) @client
  }
`
```
And when I try to generate types for **local** requests I get an error (is this probably ok?)

At the moment I only see one solution. Most likely I should separate files with local requests from requests from the `schema.graphql`.
In other words, change `--includes` which will only look at external requests.

Or do you know other ways? ...thank you in advance!

**Comment by mxswat** [comment]
Same issue here I'm planing to change my `--include` to exclude local queries

**Comment by Vlad-Shevliakov** [comment]
@mxswat  That's exactly what I did.
I put all my local queries in the apollo folder next to cache:
```
- src/
--- apollo/
----- cache/
------- queries/
------- cache.ts
...
```
In the queries folder, I left my queries, separating them by entity
But I have no idea what to do with @client in requests.
However, dubious code generation...
